src/CodeGeneration.Roslyn/CompilationGenerator.cs

```diff
@@ -157,7 +159,40 @@ public void Generate(IProgress<Diagnostic> progress = null, CancellationToken ca
         protected virtual Assembly LoadAssembly(string path)
         {
             var loadContext = AssemblyLoadContext.GetLoadContext(this.GetType().GetTypeInfo().Assembly);
-            return loadContext.LoadFromAssemblyPath(path);
+            var assembly = loadContext.LoadFromAssemblyPath(path);
+
+            var dependencyContext = DependencyContext.Load(assembly);
+
+            var assemblyResolver = new CompositeCompilationAssemblyResolver(new ICompilationAssemblyResolver[]
+            {
+                new AppBaseCompilationAssemblyResolver(Path.GetDirectoryName(path)),
+                new ReferenceAssemblyPathResolver(),
+                new PackageCompilationAssemblyResolver()
+            });
+
+            loadContext.Resolving += resolveAssembly;
```

Comment:
I don't think we should be _adding_ a new resolver every time we load an assembly. Shouldn't we define and add the resolver just once at the start of the tool? Otherwise the scaling will be awful: each assembly load will add yet another resolver and after 5 assembly loads we'll have 5 resolvers all doing equivalent things running.

Reply:
I think we need an resolver for each folder where the assembly could be. `CompositeCompilationAssemblyResolve` does not have a method to add additional resolver after construction. So I think there were two possiblitys.

1. We could use a Hashset to keep track of the direcotrys we have already an resolver and create a new one if needed.
2. We are able to get all pathes up front and only create one `CompositeCompilationAssemblyResolver` that has a resolver for every directory.

I'm not sure if number two is feasible, so I would stick to the first approach. 

Any Thoughts on this?

Reply:
I tried a third way.

Now there will actually be one callback that resolves the assembly. the used `CompositeCompilationAssemblyResolve` will actually a chain of `CompositeCompilationAssemblyResolve`s each one referencing the previous one and the directory resolver. And the new `DependencyContext` will be merged with the old.

In addition we keep track of all paths that are added to the resolver, not adding the same path twice.

Also we keep a dictionary of all loaded assembly's, if the same assembly would be loaded a second time we just return the already loaded assembly.